### PR TITLE
When making new character user can now add attributes, use cocoon gem  for dynamic display

### PR DESCRIPTION
**Pull request by StefanGajic**
Make adjustment according to comment no. 11
"https://rpg-character-editor.herokuapp.com/character_types/new - hoću da odmah mogu da definišem bilo koji broj attributes za character type koji upravo pravim. U isto vreme da mogu da napravim i char type i koliko želim attributes."

### CODE REVIEW

File: app/models/character_attribute.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class CharacterAttribute < ApplicationRecord
-  belongs_to :character_type
+  belongs_to :character_type, class_name: "CharacterType", optional: true, inverse_of: :character_attributes
```
Comment (basti) [inline app/models/character_attribute.rb]: @StefanGajic why do you need `class_name`? Rails should be able to infer it from `character_type`.

File: app/models/character_type.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 class CharacterType < ApplicationRecord
-  belongs_to :user
+  belongs_to :user, optional: true, inverse_of: :character_types
   has_one_attached :avatar
-  has_many :character_attributes, dependent: :destroy
+  has_many :character_attributes, dependent: :destroy, inverse_of: :character_type, :autosave => true
```
Comment (basti) [inline app/models/character_type.rb]: @StefanGajic use the same hash syntax, use new hash syntax, i.e. `autosave: true`.

File: app/models/character_type.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 class CharacterType < ApplicationRecord
-  belongs_to :user
+  belongs_to :user, optional: true, inverse_of: :character_types
```
Comment (basti) [inline app/models/character_type.rb]: @StefanGajic why is this optional?

**Review by basti — CHANGES_REQUESTED** [review]
@StefanGajic also your tests are failing.

File: app/models/character_attribute.rb (hunk 1 above)
Comment (StefanGajic) [reply app/models/character_attribute.rb]: I understand, I was looking at different examples and some of them had `class_name` in the model and I guess I forgot to delete it. I can see from docs that :
"If the name of the other model cannot be derived from the association name, you can use the :class_name option to supply the model name. For example, if a book belongs to an author, but the actual name of the model containing authors is Patron, you'd set things up this way: `belongs_to :author, class_name: "Patron"` "

File: app/models/character_type.rb (hunk 3 above)
Comment (StefanGajic) [reply app/models/character_type.rb]: Again I was looking at some examples, and I forgot to remove it later. I understand we don't want user to be optional

**Comment by StefanGajic** [comment]
> @StefanGajic also your tests are failing.

after changes I run all tests using `bundle exec rspec spec` and they are all passing, including the one that failed earlier

File: app/models/character_attribute.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 class CharacterAttribute < ApplicationRecord
-  belongs_to :character_type
+  belongs_to :character_type, optional: true, inverse_of: :character_attributes
```
Comment (basti) [inline app/models/character_attribute.rb]: @StefanGajic optional?

**Review by basti — APPROVED** [review]
@StefanGajic just one question for optional belongs to. Otherwise, everything else looks good.

File: app/models/character_attribute.rb (hunk 4 above)
Comment (StefanGajic) [reply app/models/character_attribute.rb]: I guess I was looking at older version of Rails again, now I see that in newer versions from late 2018 belongs_to associations are now automatically required: true

**Pull request merged**